public/pages/ShrinkIndex/container/ShrinkIndex/ShrinkIndex.tsx

```diff
@@ -0,0 +1,668 @@
+  };
+
+  shrinkIndex = async (sourceIndexName: string, targetIndexName: string, requestPayload: Required<IndexItem>["settings"]) => {
+    try {
+      const { commonService } = this.props;
+      const { aliases, ...settings } = requestPayload;
+
+      const result = await commonService.apiCaller({
+        endpoint: "indices.shrink",
+        data: {
+          index: sourceIndexName,
+          target: targetIndexName,
+          body: {
+            settings: {
+              ...settings,
+            },
+            aliases,
+          },
+        },
+      });
+      if (result && result.ok) {
+        const toastInstance = this.context.notifications.toasts.addSuccess(
+          `Successfully started shrinking ${sourceIndexName}. The shrunken index will be named ${targetIndexName}.`,
+          {
+            toastLifeTimeMs: 1000 * 60 * 60 * 24 * 5,
+          }
+        );
+        this.onCancel();
+        jobSchedulerInstance.addJob({
+          interval: 30000,
+          extras: {
+            toastId: toastInstance.id,
+            sourceIndex: sourceIndexName,
+            destIndex: targetIndexName,
+          },
+          type: "shrink",
+        } as RecoveryJobMetaData);
+      } else {
+        this.context.notifications.toasts.addDanger(result.error);
+      }
+    } catch (err) {
+      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem shrinking index."));
+    }
+  };
+
+  getIndexSettings = async (indexName: string, flat: boolean): Promise<Record<string, IndexItem> | void> => {
+    try {
+      const { commonService } = this.props;
+      const result: ServerResponse<Record<string, IndexItem>> = await commonService.apiCaller({
+        endpoint: "indices.getSettings",
+        data: {
+          index: indexName,
+          flat_settings: flat,
+        },
+      });
+      if (result && result.ok) {
+        return result.response;
+      } else {
+        const errorMessage = `There is a problem getting index setting for ${indexName}, please check with Admin.`;
+        this.context.notifications.toasts.addDanger(result?.error || errorMessage);
+      }
+    } catch (err) {
+      this.context.notifications.toasts.addDanger(getErrorMessage(err, "There was a problem getting index settings."));
+    }
+  };
+
+  setIndexSettings = async (indexName: string, settings: {}) => {
+    try {
+      const { commonService } = this.props;
+      const result = await commonService.apiCaller({
+        endpoint: "indices.putSettings",
+        method: "PUT",
+        data: {
+          index: indexName,
+          body: {
+            settings: {
+              ...settings,
+            },
+          },
+        },
+      });
+      if (result && result.ok) {
+        this.context.notifications.toasts.addSuccess(`${indexName} has been set to block write operations.`);
+      } else {
+        const errorMessage = `There is a problem set index setting for ${indexName}, please check with Admin`;
```

Comment:
There seems to be some typo here
`There is a problem set index setting for`. Should it be `There is a problem when updating index setting for...`

Reply:
Yeah, thanks for your review, I'll change it and check the whole file.